gateway/service/src/main/java/org/eclipse/ditto/gateway/service/endpoints/routes/RootRouteHeadersStepBuilder.java

```diff
@@ -122,7 +124,16 @@ QueryParametersStep withRequestContext(final RequestContext requestContext) {
 
         private Map<String, String> getFilteredExternalHeaders(final HttpMessage httpRequest) {
             final Iterable<HttpHeader> headers = httpRequest.getHeaders();
+            final Set<String> headerNames = new HashSet<>();
             final Map<String, String> externalHeaders = StreamSupport.stream(headers.spliterator(), false)
+                    .map(httpHeader -> {
```

Comment:
Just out of curiocity, what is the reason behind this double duplicate headers check?